test/suites/dev/common/test-eip7702/test-eip7702-delegatecall.ts

```diff
@@ -0,0 +1,444 @@
+        const targetEOA = (await createFundedAccount(context)).account;
+
+        // Create authorization
+        const authorization = await delegatingEOA.signAuthorization({
+          contractAddress: callerAddress,
+          chainId: chainId,
+          nonce: 0,
+        });
+
+        // Delegatecall to EOA
+        const callData = encodeFunctionData({
+          abi: callerAbi,
+          functionName: "delegatecallAddress",
+          args: [targetEOA.address, "0x"],
+        });
+
+        const tx = {
+          to: delegatingEOA.address,
+          data: callData,
+          chainId: chainId,
+          authorizationList: [authorization],
+          txnType: "eip7702" as const,
+          privateKey: sender.privateKey,
+        };
+
+        const signedTx = await createViemTransaction(context, tx);
+        const hash = await sendRawTransaction(context, signedTx);
+        await context.createBlock();
+
+        const receipt = await context.viem().getTransactionReceipt({
+          hash,
+        });
+        expect(receipt.status).toBe("success");
+      },
+    });
+
+    it({
+      id: "T03",
+      title: "should perform delegatecall to contract account",
+      test: async () => {
+        const sender = await createFundedAccount(context);
+        const delegatingEOA = (await createFundedAccount(context)).account;
+
+        // Create authorization for caller contract
+        const authorization = await delegatingEOA.signAuthorization({
+          contractAddress: callerAddress,
+          chainId: chainId,
+          nonce: 0,
+        });
+
+        // Delegatecall to storage writer contract
+        const storeData = encodeFunctionData({
+          abi: storageWriterAbi,
+          functionName: "store",
+          args: [1n, 42n],
+        });
+
+        // NOTE: When contract A executes delegatecall to contract B, B's code is executed
+        // with contract A's storage, msg.sender and msg.value
+        const callData = encodeFunctionData({
+          abi: callerAbi,
+          functionName: "delegatecallAddress",
+          args: [storageWriterAddress, storeData],
+        });
+
+        const tx = {
+          to: delegatingEOA.address,
+          data: callData,
+          gas: 1_500_000n,
+          chainId: chainId,
+          authorizationList: [authorization],
+          txnType: "eip7702" as const,
+          privateKey: sender.privateKey,
+        };
+
+        const signedTx = await createViemTransaction(context, tx);
+        const hash = await sendRawTransaction(context, signedTx);
+        await context.createBlock();
+
+        const receipt = await context.viem().getTransactionReceipt({
+          hash,
+        });
+        expect(receipt.status).toBe("success");
+
+        // Storage should be in the delegating EOA's context (via caller contract delegation)
+        // This is complex because of double delegation - may need to check actual storage slots
```

Comment:
This seems like a leftover `TODO`

Reply:
Fixed in db23977eaa744ca16134688749bf77ed89e50949